content/chainguard/chainguard-images/chainguard-registry/pull-through-guides/artifactory/artifactory-packages-pull-through/index.md

```diff
@@ -171,43 +203,90 @@ cat > Dockerfile.build <<EOF
 
-<center><img src="ar-apk-5.png" alt="Screenshot showing a portion of the 'New Virtual Repository' page. The 'cg-wolfi' and 'cg-extras' repositories have been checked and moved to the 'Selected Repositories' column." style="width:750px;"></center>
-<br />
+```shell
+export CG_TOKEN=<cg-chainguard-token>
+```
+
+Then for the `cg-extras` repo's token:
+
+```shell
+export EXTRAS_TOKEN=<cg-extras-token>
+```
 
-Finally, click the **Create Virtual Repository** button. As before, a modal window will appear letting you know that the repository was created successfully. Click the **Set Up Generic Client** button at the bottom of this window to retrieve the token you'll need to test whether you can pull images through this repository. 
+Following that, you can use these environment variables, along with a few created earlier in the guide, to create a Dockerfile as you previously did for the `cg-private` repository:
+
+```shell
+cat > Dockerfile.repos <<EOF
+FROM cgr.dev/chainguard/python:latest-dev
+USER root
+RUN mv /etc/apk/repositories /etc/apk/repositories.disabled
+RUN echo 'https://$ARTIFACTORY_USER_PROFILE:$CG_TOKEN@$ARTIFACTORY_HOSTNAME.jfrog.io/artifactory/cg-chainguard/' >> /etc/apk/repositories
+RUN echo 'https://$ARTIFACTORY_USER_PROFILE:$EXTRAS_TOKEN@$ARTIFACTORY_HOSTNAME.jfrog.io/artifactory/cg-extras/' >> /etc/apk/repositories
+RUN apk add c-ares
+EOF
+```
+
+Following that, build the image:
+
+```shell
+docker build -t ar-repos -f Dockerfile.repos .
+```
+
+As this command runs, it will install the `c-ares` package into the image.
+
+
+## Using a Virtual Repository as a Pull-Through Cache
+
+Artifactory allows you to create what it refers to as *virtual repositories*. A virtual repository is a collection of one or more repositories (such as local, remote, or other virtual repositories) that have the same package type. The benefit of this is that you can access resources from multiple locations using a single logical URL.
+
+> **Note**: This section provides a high-level overview of how to set up a virtual repository on Artifactory that you can use as a pull-through cache for Chainguard packages. For more information on Artifactory's virtual repositories, please refer to the [official documentation](https://jfrog.com/help/r/jfrog-artifactory-documentation/virtual-repositories).
+
+To get started, create a new virtual repository. From the **Repositories** tab, click the **Create a Repository** button. This time, select the **Virtual** option, but instead of selecting the Alpine package type, select **Generic**.
+
+You can use the Alpine package type, but [virtual Alpine repositories](https://jfrog.com/help/r/jfrog-artifactory-documentation/virtual-alpine-repositories) will only allow you to pull packages that have already been cached by an associated remote repository. A **Generic** artifact repository will provide more flexibility.
```

Comment:
Really .. that sounds like a bug .... I am not reading that from their docs.. how did you get to that info?

Reply:
That was already in the doc, so wherever it came form I got this detail a long time ago. With some testing just now, though, it seems to be true so I'm inclined to leave it in.